src/xcode/ENA/ENA/Source/Models/Exposure/ExposureManager.swift

```diff
@@ -207,9 +208,21 @@ final class ENAExposureManager: NSObject, ExposureManager {
 	/// Wrapper for `ENManager.detectExposures`
 	/// `ExposureManager` needs to be activated and enabled
 	func detectExposures(configuration: ENExposureConfiguration, diagnosisKeyURLs: [URL], completionHandler: @escaping ENDetectExposuresHandler) -> Progress {
-		manager.detectExposures(configuration: configuration, diagnosisKeyURLs: diagnosisKeyURLs) { summary, error in
+
+		// An exposure detection is currently running. Return current progress.
+		if let progress = progress, !progress.isCancelled && !progress.isFinished {
```

Comment:
@ArturFriesen did you also verify that this works through manual tests? IIRC, the `Progress` object returned by ENF is not fully functional, i.e. we don't get a real progress of the exposure detection (which we do not need here), I think calling `cancel()` to interrupt the exposure detection is not supported and a few other strange behaviors. So I'm just wondering if `isCancelled` and `isFinished` work reliably on a real device.

Reply:
Yes I tested it on device. Basically it is working with the optionality of progress. It is set to nil on completion.
Additionally as far as I know, cancel() works and needs to be taken into account here. Christian fixed a bug by calling cancel() on timeout, see [here](https://github.com/corona-warn-app/cwa-app-ios/blob/e022c608cc4512019d581a811225fef1d2182a44/src/xcode/ENA/ENA/Source/Services/Risk/Provider/RiskProvider.swift#L280) . "isFinished" is not working, because the progress is not tracked inside of ENF. I have added it for the case it could start work in the future.